packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -1,32 +1,1237 @@
+
+  /**
+   * The method to match on.
+   *
+   * @default - no match on method
+   */
+  readonly method?: Method,
+
+  /**
+   * The prefix or path to match on
+   *
+   * @default - match on '/'
+   */
+  readonly prefixOrPath?: HttpGatewayRoutePathOrPrefixMatch,
+}
+
+/**
+ * Configuration for HTTP gateway route path match
+ */
+export interface HttpPathMatchConfig {
+  /**
+   * GatewayRoute CFN configuration for HTTP gateway route path match.
+   */
+  readonly pathMatch: CfnGatewayRoute.HttpPathMatchProperty;
+}
+
+/**
+ * Used to generate HTTP path matching methods.
+ */
+export abstract class HttpPathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   *
+   * @param path The exact path to match on
+   */
+  public static matchingExactly(path: string): HttpPathMatch {
+    return new HttpPathMatchImpl({ exact: path });
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   * @param regex The regex used to match the path.
+   */
+  public static matchingRegex(regex: string): HttpPathMatch {
+    return new HttpPathMatchImpl({ regex: regex });
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  abstract bind(scope: Construct): HttpPathMatchConfig;
+}
+
+class HttpPathMatchImpl extends HttpPathMatch {
+  constructor(
+    private readonly pathMatch: CfnGatewayRoute.HttpPathMatchProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpPathMatchConfig {
+    return {
+      pathMatch: this.pathMatch,
+    };
+  }
+}
+
+/**
+ * Configuration for `QueryParameterMatch`
+ */
+export interface QueryParameterConfig {
+  /**
+   * The HTTP route header.
+   */
+  readonly queryParameter: CfnGatewayRoute.QueryParameterProperty;
+}
+
+/**
+ * Used to generate query parameter matching methods.
+ */
+export abstract class QueryParameterMatch {
+  /**
+   * The value of the query parameter with the given name in the request must match the
+   * specified value exactly.
+   *
+   * @param queryParameterName the name of the query parameter to match against
+   * @param queryParameterValue The exact value to test against
+   */
+  static valueIs(queryParameterName: string, queryParameterValue: string): QueryParameterMatch {
+    return new QueryParameterMatchImpl(queryParameterName, { exact: queryParameterValue });
+  }
+
+  /**
+   * Returns the query parameter match configuration.
+   */
+  abstract bind(scope: Construct): QueryParameterConfig;
+}
+
+class QueryParameterMatchImpl extends QueryParameterMatch {
+  constructor(
+    private readonly queryParameterName: string,
+    private readonly matchProperty: CfnGatewayRoute.HttpQueryParameterMatchProperty,
+  ) {
+    super();
+  }
+
+  bind(_scope: Construct): QueryParameterConfig {
+    return {
+      queryParameter: {
+        match: this.matchProperty,
+        name: this.queryParameterName,
+      },
+    };
+  }
+}
+
+/**
+ * Enum of method
+ */
+export enum Method {
```

Comment:
Do we have a shared `Method` enum?

Reply:
Great call out. `Route` already defined the method.

I might created a shared file between `Route` and `GatewayRoute` and move the method there.